kong/plugins/aws-lambda/iam-eks-credentials.lua

```diff
@@ -0,0 +1,77 @@
+local kong = kong
+local ENV_TOKEN_FILE = os.getenv 'AWS_WEB_IDENTITY_TOKEN_FILE'
+local ENV_ROLE_ARN = os.getenv 'AWS_ROLE_ARN'
+local DEFAULT_SERVICE_REQUEST_TIMEOUT = 5000
+
+
+local url = require "socket.url"
+local http = require "resty.http"
+local json = require "cjson"
+local ngx_now = ngx.now
+local concat = table.concat
+local tostring = tostring
+
+do
+  if not (ENV_TOKEN_FILE) then
+    -- No variables found, so we're not probably not running on EKS containers
+    kong.log.debug("No Web Identity environment variables found for IAM")
+  end
+end
```

Comment:
Idea: if we already know at `require`-time that no `AWS_WEB_IDENTITY_TOKEN_FILE` is set, then we might as well exit early:

```suggestion

if not ENV_TOKEN_FILE or not ENV_ROLE_ARN then
  -- No variables found, so we're not probably not running on EKS containers
  kong.log.debug("No Web Identity environment variables found for IAM")
  return { configured = false }
end

local url = require "socket.url"
local http = require "resty.http"
local json = require "cjson"
local ngx_now = ngx.now
local concat = table.concat
local tostring = tostring
```

Alternatively, you could also invert the logic:

```lua
local configured = false
local fetchCredentials

if ENV_TOKEN_FILE and ENV_ROLE_ARN then
  configured = true

  local url = require "socket.url"
  local http = require "resty.http"
  -- [...]

  function fetchCredentials()
    --  [...]
  end
end

return {
  configured = configured,
  fetchCredentials = fetchCredentials,
}
```